src/Mvc/Mvc.ViewFeatures/src/Buffers/ViewBufferTextWriter.cs

```diff
@@ -86,25 +86,20 @@ public ViewBufferTextWriter(ViewBuffer buffer, Encoding encoding, HtmlEncoder ht
         /// <inheritdoc />
         public override Encoding Encoding { get; }
 
-        /// <inheritdoc />
-        public bool IsBuffering { get; private set; } = true;
-
         /// <summary>
         /// Gets the <see cref="ViewBuffer"/>.
         /// </summary>
         public ViewBuffer Buffer { get; }
 
+        /// <summary>
+        /// Gets a value that indiciates if <see cref="Flush"/> or <see cref="FlushAsync" /> was invoked.
+        /// </summary>
+        public bool HasFlushed { get; private set; }
+
         /// <inheritdoc />
         public override void Write(char value)
         {
-            if (IsBuffering)
```

Comment:
Some background - `ViewBufferTextWriter` had a dual mode of operation - after a `Flush`, all writes directly went to the underlying `TextWriter` (i.e. `HttpResponseStreamWriter`). With large enough content, we'll run out of buffer in the writer and result in a synchronous write that throws. 

There isn't a very good reason for us to stop buffering once a flush is performed, it's really not an important \ well-defined requirement of `FlushAsync` .

Reply:
Yeah I agree. This seems like such an obvious improvment, I'm sad we didn't think of it befores.